tools/Repo-Tasks.psm1

```diff
@@ -0,0 +1,245 @@
+﻿$taskScriptDir = [System.IO.Path]::GetDirectoryName($PSCommandPath)
+$env:repoRoot = [System.IO.Path]::GetDirectoryName($taskScriptDir)
+$userPsFileDir = [string]::Empty
+
+[string]$envVariableName="TEST_CSM_ORGID_AUTHENTICATION"
+
+[CmdletBinding]
+Function Set-TestEnvironment
+{
+<#
+.SYNOPSIS
+This cmdlet helps you to setup Test Environment for running tests
+In order to successfully run a test, you will need SubscriptionId, TenantId
+This cmdlet will only prompt you for Subscription and Tenant information, rest all other parameters are optional
+
+#>
+    [CmdletBinding(DefaultParameterSetName='UserIdParamSet')]
+    param(
+        [Parameter(ParameterSetName='UserIdParamSet', Mandatory=$true, HelpMessage = "UserId (OrgId) you would like to use")]
+        [ValidateNotNullOrEmpty()]
+        [string]$UserId,
+
+        [Parameter(ParameterSetName='UserIdParamSet', Mandatory=$true, HelpMessage = "UserId (OrgId) you would like to use")]
+        [ValidateNotNullOrEmpty()]
+        [string]$Password,
+
+        [Parameter(ParameterSetName='SpnParamSet', Mandatory=$true, HelpMessage='ServicePrincipal/ClientId you would like to use')]   
+        [ValidateNotNullOrEmpty()]
+        [string]$ServicePrincipal,
+
+        [Parameter(ParameterSetName='SpnParamSet', Mandatory=$true, HelpMessage='ServicePrincipal Secret/ClientId Secret you would like to use')]
+        [ValidateNotNullOrEmpty()]
+        [string]$ServicePrincipalSecret,
+
+        [Parameter(ParameterSetName='SpnParamSet', Mandatory=$true)]
+        [Parameter(ParameterSetName='UserIdParamSet', Mandatory=$true, HelpMessage = "SubscriptionId you would like to use")]
+        [ValidateNotNullOrEmpty()]
+        [string]$SubscriptionId,
+
+        [Parameter(ParameterSetName='SpnParamSet', Mandatory=$true, HelpMessage='AADTenant/TenantId you would like to use')]
+        [ValidateNotNullOrEmpty()]
+        [string]$TenantId,
+
+        [ValidateSet("Playback", "Record", "None")]
+        [string]$RecordMode='Playback',
+
+        [ValidateSet("Prod", "Dogfood", "Current", "Next")]
+        [string]$TargetEnvironment='Prod'
+    )
+
+    [string]$uris="https://management.azure.com/;AADAuthEndpoint=https://login.windows.net/;GraphUri=https://graph.windows.net/"
```

Comment:
Only the BaseUri is needed in this case, the rest of these (AADAuthENdpoint, GraphUri) will override whatever values are in Environment, so we should remove these